Gameplay/misc/LaserController.cpp

```diff
@@ -0,0 +1,188 @@
+#include "scriptingUtil/gameplaypch.h"
+#include "LaserController.h"
+#include "constants/Scenes.h"
+#include "entities/player/PlayerController.h"
+
+//#include "components/ComponentObstacle.h"
+
+// TODO: Delete this include:
+#include <modules/ModuleSceneManager.h>
+
+Hachiko::Scripting::LaserController::LaserController(GameObject* game_object)
+	: Script(game_object, "LaserController")
+	, _laser(nullptr)
+	, _max_length(5.0f)
+	, _max_scale(0.2f)
+	, _activation_time(1.0f)
+	, _damage(1.0f)
+	, _spin_movement(false)
+	, _spin_clockwise(true)
+	, _spin_speed(20.0f)
+	, _horizonal_movement(false)
+	, _movement_speed(1.0f)
+	, _movement_target(nullptr)
+	, _toggle_activation(false)
+	, _toggle_active_time(1.0f)
+	, _toggle_inactive_time(1.0f)
+{}
+
+void Hachiko::Scripting::LaserController::OnAwake()
+{
+	_terrain = Scenes::GetTerrainContainer();
+	_player = Scenes::GetPlayer();
+	//door_obstacle = _door->GetComponent<ComponentObstacle>();
```

Comment:
```suggestion
```

Let's delete this maybe? Because it's not related with the Script itself

Reply:
++